lib/evaluate/check-call.h

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Constraint checking for procedure references
+
+#ifndef FORTRAN_EVALUATE_CHECK_CALL_H_
+#define FORTRAN_EVALUATE_CHECK_CALL_H_
+
+#include "call.h"
+#include "characteristics.h"
+#include "expression.h"
+#include <optional>
```

Comment:
Not needed?

Reply:
Replaced with forward references to classes.